TestingLowerBounds/ForMathlib/LeftRightDeriv.lean

```diff
@@ -0,0 +1,190 @@
+/-
+Copyright (c) 2024 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne, Lorenzo Luccioli
+-/
+import TestingLowerBounds.Testing.Binary
+import TestingLowerBounds.FDiv.Basic
+import Mathlib.Analysis.Calculus.Monotone
+import Mathlib.Analysis.Convex.Deriv
+import TestingLowerBounds.ForMathlib.MonotoneOnTendsto
+
+
+open Set Filter Topology
+
+open scoped ENNReal NNReal
+
+noncomputable
+def rightDeriv (f : ℝ → ℝ) : ℝ → ℝ := fun x ↦ derivWithin f (Ioi x) x
```

Comment:
Let's add docstrings to the definitions (and more generally, run `#lint` on the files).
(I know I am guilty of pushing several defs without docstrings... but let's improve on that).